### PR TITLE
🎉Source Twilio: New stream Trunks

### LINKED ISSUE #19746: Source Twilio: new stream Trunk
## Tell us about the problem you're trying to solve
The Trunk resource represents a SIP Trunk in your Twilio account. A Trunk can have associated [Credential Lists](https://www.twilio.com/docs/sip-trunking/api/credentiallist-resource), [IP Access Control Lists](https://www.twilio.com/docs/sip-trunking/api/ipaccesscontrollist-resource), [Origination URLs](https://www.twilio.com/docs/sip-trunking/api/originationurl-resource), and [Phone Numbers](https://www.twilio.com/docs/sip-trunking/api/phonenumber-resource).
GET https://trunking.twilio.com/v1/Trunk

## Describe the solution you’d like
Research pros and cons for implemnting this stream

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes #19746

## Enhancement
New streams: 

- Trunks

## Recommended reading order
1. `source_twilio/source.py`
2. `source_twilio/streams.py`
3.  `source_twilio/schemas/trunks.json`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter [Yes]

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by sh4sh** [comment]
Note: before merging, we should bump the version in the connector's metadata.yaml as part of the new connector image publishing process

**Comment by sh4sh** [comment]
Seeded the integration test account for the new trunks stream, tests should be better now

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev I'd just need the final approval from our connector team to release your code.

File: airbyte-integrations/connectors/source-twilio/source_twilio/streams.py
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-twilio/source_twilio/streams.py:28]: Need add to allowedHosts

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-twilio/source_twilio/streams.py
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-twilio/source_twilio/streams.py:28]: Good catch.

**Pull request merged**
